http-clients/src/test/java/org/triplea/http/client/web/socket/WebSocketConnectionTest.java

```diff
@@ -52,39 +55,71 @@ void tearDown() {
 
     @Test
     void onMessage() {
-      webSocketConnection.getClient().onMessage(MESSAGE);
+      webSocketConnection.getWebSocketListener().onText(mock(WebSocket.class), MESSAGE, true);
       verify(webSocketConnectionListener).messageReceived(MESSAGE);
     }
 
+    @Test
+    void verifyListenerAccumulatesMessagesUntilLast() {
```

Comment:
I think we may need some additional test cases that exercise concurrency with multiple, broken up messages. I'm concerned about that scenario.

Reply:
As I already mentioned, the JDK API won't invoke the listener again unless we call `webSocket.request(1 or higher)`.
From [the documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/WebSocket.html):
>  A receive method is any of the onText, onBinary, onPing, onPong and onClose methods of Listener. WebSocket initiates a receive operation by invoking a receive method on the listener. The listener then must return a CompletionStage which completes once the operation has completed.
>
>To control receiving of messages, a WebSocket maintains an internal counter. This counter's value is a number of times the WebSocket has yet to invoke a receive method. While this counter is zero the WebSocket does not invoke receive methods. The counter is incremented by n when request(n) is called. The counter is decremented by one when the WebSocket invokes a receive method.

Also the [WebSocket.Listener documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/WebSocket.Listener.html) states:
> The strict sequential order of invocations from WebSocket to Listener means, in particular, that the Listener's methods are treated as non-reentrant. This means that Listener implementations do not need to be concerned with possible recursion or the order in which they invoke WebSocket.request in relation to their processing logic.
>
>Careful attention may be required if a listener is associated with more than a single WebSocket. In this case invocations related to different instances of WebSocket may not be ordered and may even happen concurrently. 


So as long as we're not re-using a listener for multiple websockets we are fine